Test/MetaDrawTest.cs

```diff
@@ -54,5 +55,40 @@ public static void TestMetaDrawReadCrossPsmFile()
 
             Directory.Delete(folderPath, true);
         }
+
+        [Test]
+        public static void TestLabelingVariantCrossingIons()
+        {
+            string myFile = Path.Combine(TestContext.CurrentContext.TestDirectory, @"TestData\VariantCrossTest.psmtsv");
+            List<string> warnings = new List<string>();
+            List<PsmFromTsv> psms;
+
+            psms = PsmTsvReader.ReadTsv(myFile, out warnings);  //test will fail if order of psms isn't from top to bottom row
+
+            List<Dictionary<string, bool>> expected = new List<Dictionary<string, bool>>();
+            expected.Add(new Dictionary<string, bool>() { { "b2", false }, { "b16", false }, { "y3", false }, { "y35", false } });   // no variant (7527)
+            expected.Add(new Dictionary<string, bool>() { { "b2", false }, { "b3", true } });                                        // b fragments before and after variant (4221)
+            expected.Add(new Dictionary<string, bool>() { { "y2", false }, { "y3", true } });                                        // y fragments before and after variant (7759)
+            expected.Add(new Dictionary<string, bool>() { { "b4", true }, { "b16", true }, { "y1", false }, { "y7", false } });      // variant at 1st position (9221)
+            expected.Add(new Dictionary<string, bool>() { { "b2", false }, { "b3", false }, { "y1", true }, { "y9", true } });       // variant at last position (6778)
+            expected.Add(new Dictionary<string, bool>() { { "b3", false }, { "b4", true }, { "y34", false }, { "y35", true } });     // peptide comes from multiple proteins (8613)
+            expected.Add(new Dictionary<string, bool>() { { "b1", true }, { "b10", true }, { "y1", true }, { "y6", true } });        // variation spans the whole peptide (8765)
+            expected.Add(new Dictionary<string, bool>() { { "b2", false }, { "b5", false }, { "y1", false }, { "y30", false } });    // variant before peptide (8169)
+            expected.Add(new Dictionary<string, bool>() { { "b2", false }, { "b4", false }, { "y1", false }, { "y14", false } });    // variant after peptide (6532)
+
+
+            Assert.AreEqual(0, psms[0].VariantCrossingIons.Count, 
```

Comment:
Don't just check for one psm. Ensure that the count is as expected for all the psms

Reply:
The whole unit test works differently now that VariantCrossingIons is a List, but it checks that there aren't any extra ions in the list.